### ISSUE TITLE
Avoid computing transition tables used for Uri scanning at runtime, instead precompute them and make corelibs use a constant

### ISSUE THREAD

**Issue by mkustermann** [dart-lang/sdk#49108]
As part of starting an isolate we run bootstrapping code. Part of that will use the Uri class which triggers lazy creation of Uri scanning tables, see [sdk/lib/core/uri.dart](https://github.com/dart-lang/sdk/blob/362afe3656c828/sdk/lib/core/uri.dart#L4037):
```dart
final List<Uint8List> _scannerTables = _createTables();

List<Uint8List> _createTables() {
  // TODO(lrn): Use a precomputed table.
  ... lots of logic here ...
}
```

=> This is not good for startup latency in general.
=> It is a base cost for any `Isolate.spawn()`. Making it less appealing to use isolates for short-lived tasks.

The tables can be pre-computed and baked in as a constant into the core libraries, addressing the existing todo in the code.

/cc @lrhn

**Comment by lrhn** [comment]
I was planning to make the table a single string, but stopped for some reason that I can no longer remember.
(Could be that we were considering a rewrite of `Uri` for Dart 2.0 anyway, it just didn't make the final list.)

**Comment by kevmoo** [comment]
Fix reverted in 855e1cd975feae07822187f226568130a0fcfbe9

**Comment by lrhn** [comment]
Relanded, and appears to have stuck.
